src/main/kotlin/ar/edu/unahur/obj2/servidorWeb/ServidorWeb.kt

```diff
@@ -10,25 +10,42 @@ enum class CodigoHttp(val codigo: Int) {
 
 class ServidorWeb {
   val modulos = mutableListOf<Modulo>()
+  val analizadores = mutableListOf<Analizador>()
 
   fun realizarPedido(ip: String, url: String, fechaHora: LocalDateTime): Respuesta {
+    // Creamos el pedido
+    val pedidoRealizado = Pedido(ip = ip, url = url, fechaHora = fechaHora)
     if (!url.startsWith("http:")) {
-      return Respuesta(codigo = CodigoHttp.NOT_IMPLEMENTED, body = "", tiempo = 10)
+      // Le pasamos la respuesta a todos los analizadores que contenga el servidor
+      analizadores.forEach { analizador -> analizador.analizar(Respuesta(codigo = CodigoHttp.NOT_IMPLEMENTED, body = "", tiempo = 10, pedido = pedidoRealizado)) }
+      return Respuesta(codigo = CodigoHttp.NOT_IMPLEMENTED, body = "", tiempo = 10, pedido = pedidoRealizado)
     }
-
     if (this.algunModuloSoporta(url)) {
       val moduloSeleccionado = this.modulos.find { it.puedeTrabajarCon(url) }!!
-      return Respuesta(CodigoHttp.OK, moduloSeleccionado.body, moduloSeleccionado.tiempoRespuesta)
+      // Le pasamos la respuesta a todos los analizadores que contenga el servidor
+      analizadores.forEach { analizador -> analizador.analizar(Respuesta(CodigoHttp.OK, moduloSeleccionado.body, moduloSeleccionado.tiempoRespuesta, pedidoRealizado, modulo = moduloSeleccionado))}
+      return Respuesta(CodigoHttp.OK, moduloSeleccionado.body, moduloSeleccionado.tiempoRespuesta, pedidoRealizado)
     }
-
-    return Respuesta(codigo = CodigoHttp.NOT_FOUND, body = "", tiempo = 10)
+    // Le pasamos la respuesta a todos los analizadores que contenga el servidor
+    analizadores.forEach { analizador -> analizador.analizar(Respuesta(codigo = CodigoHttp.NOT_FOUND, body = "", tiempo = 10, pedido = pedidoRealizado))}
+    return Respuesta(codigo = CodigoHttp.NOT_FOUND, body = "", tiempo = 10, pedido = pedidoRealizado)
   }
 
   fun algunModuloSoporta(url: String) = this.modulos.any { it.puedeTrabajarCon(url) }
-
   fun agregarModulo(modulo: Modulo) {
     this.modulos.add(modulo)
+
+  }
+
+  fun agregarAnalizador(analizador: Analizador){
+    analizadores.add(analizador)
+  }
+
+  fun sacarAnalizador(analizador: Analizador){
+    analizadores.add(analizador)
   }
 }
 
-class Respuesta(val codigo: CodigoHttp, val body: String, val tiempo: Int)
+class Respuesta(val codigo: CodigoHttp, val body: String, val tiempo: Int, val pedido: Pedido, val modulo:Modulo? = null) {}
+
+class Pedido(val ip: String, val url: String, val fechaHora: LocalDateTime){}
```

Comment:
Las llaves finales no son necesarias si no vas a definir ningún método. Quedaría así:
```suggestion
class Pedido(val ip: String, val url: String, val fechaHora: LocalDateTime)
```